hypertools/plot/plot.py

```diff
@@ -230,11 +230,16 @@ def plot(x, fmt=None, marker=None, markers=None, linestyle=None,
         x = reducer(x, ndims=3, reduce=reduce, internal=True)
 
     # find cluster and reshape if n_clusters
-    if n_clusters is not None:
-        cluster_labels = clusterer(x, cluster=cluster, n_clusters=n_clusters)
-        x = reshape_data(x, cluster_labels)
+    if cluster is not None:
+        cluster_labels= clusterer(x,cluster=cluster,n_clusters=n_clusters) if n_clusters is not None else clusterer(x,cluster=cluster)
+        if is_label_probabilistic(cluster_labels):
+            cluster_discrete_labels=reshape_labels(cluster_labels)
+            x = reshape_data(x, cluster_discrete_labels)
```

Comment:
i think we should reshape the data into the number of unique values returned by `transform_labels2RGB`.  If its too comp expensive to plot in that format, we can set the max number of bins to be maybe 100 or so

Reply:
I think number of bins should have an upper cap. Because with possibilities being a continuous value we may end up with a very large number of bins- theoretically equal to number of input data points or samples. Hence, we can't afford reshaping the data into the number of unique values returned. [Let me know, if I got it wrong here!].
Assuming the above hypothesis to be valid, we can think of below alternatives:

Option1: As per your suggestion, Lets put an upper limit on maximum number of bins(say 100) to be plotted. If you observe - It becomes a clustering problem again! So, we take all these points and assign them to the pool of 100 Clusters, and the clusters mean color is assigned to each cluster member.
Option 2: Use Scatter plot where each plot can take a colour argument. I am not sure how does it fit with the overall aesthetics. I had used in my commit for a quick iteration.
Option 3: Call the Plot (n-1) times, where n is total number of data points; This way while drawing each line I can assign the colour. This option might not be great basis the efficiency. Also, A small line segment colour should also change basis it's proximity to the end-points.
My personal ordering would be: Option1(Most Preferable)>Option2>Option3(Least Preferable).
Let me know, if I am missing anything.

Reply:
I agree that we should bin (option 1). We could expose a `resolution` (or `res`) parameter in the plot function that defaults to 100, but can be changed by the user for finer control.  

Note that while we are now assigning the points to a fixed number of bins, the colors represent the weighted average across a number of clusters. This is different than normal discrete clustering. For example take two point clouds and a series of points between them:

```
# simulate clusters
cluster1 = np.random.multivariate_normal(np.zeros(3)-10, np.eye(3), size=100)
cluster2 = np.random.multivariate_normal(np.zeros(3)+10, np.eye(3), size=100)
intervening = np.array([[i, i, i] for i in range(-10,10)])
data = np.vstack([cluster1, intervening, cluster2])

# get cluster labels
cluster_labels = hyp.cluster(data, n_clusters=2)

# plot
hyp.plot(data, 'o', group=cluster_labels)
```
![image](https://user-images.githubusercontent.com/6596567/34153615-b270b2c2-e480-11e7-816f-c0af26713bf1.png)

The mixture models will color the intervening points as weighted averages of the two clusters, whereas the other models will treat them as discrete clusters. This behavior will be different than just specifying more dicrete clusters:

```
# get cluster labels
cluster_labels = hyp.cluster(data, n_clusters=10)

# plot
hyp.plot(data, 'o', group=cluster_labels)
```

![image](https://user-images.githubusercontent.com/6596567/34153711-042bbd6e-e481-11e7-814a-1d4195ece7c0.png)

Reply:
@jeremymanning can you weigh in on this?

Reply:
Latest commit achieves the required behaviour.  
I have hard-coded the value of `RESOLUTION=4` at line 236 in `plot.py` - This will make things difficult for you to play with the visualization. 
Idea was to quickly get your feedback and if it works then to expose this as a parameter to the user. Let me know, if you need it to be exposed for your evaluation.

Reply:
resolution is now exposed as a parameter.